### ISSUE TITLE
Support ScyllaDB JDBC as plugin

### ISSUE THREAD

**Issue by robd003** [trinodb/trino#10276]
It would be great to add the ScyllaDB JDBC driver to the available plugins since it would really decrease load on Scylla clusters compared to the generic Cassandra driver.

https://github.com/scylladb/java-driver/tree/scylla-4.x

Cassandra JDBC client on the left, Scylla JDBC client on the right:
![image](https://user-images.githubusercontent.com/571328/145689543-aeb4dfc6-23fb-4fe0-b62b-6431726488e9.png)

**Comment by robd003** [comment]
One thing that should make this *easy* is that the existing Trino Cassandra JDBC plugin used is from datastax and the Scylla JDBC is just a modified version of the datastax driver that is shard aware.

**Comment by helenap** [comment]
I need this

**Comment by robd003** [comment]
Here's where you can find the latest Scylla JDBC drivers: https://mvnrepository.com/artifact/com.scylladb/java-driver-core

**Comment by ebyhr** [comment]
We are still using version 3.x in Cassandra connector for the backward compatibility. That is why I upgraded 3.10.2 in `trinodb/trino-cassandra-driver`.

**Comment by robd003** [comment]
I'm suggesting we make an entirely new plugin trino-scylla that will use the latest connector from scylla and stay up to date with that.

The existing cassandra plugin can stay at 3.x and won't affect any legacy clusters running cassandra 2.x

**Comment by ebyhr** [comment]
If we create a new ScyllaDB connector without reusing Cassandra connector, it will take few months until release in my estimation. Actually, the library isn't JDBC driver and we need to write many codes. Is version 4.x required in your environment?

**Comment by robd003** [comment]
I'm using Scylla 4.5 at the moment. The 4.x driver should be compatible with Cassandra 2.1 up through the current release.

**Comment by ebyhr** [comment]
Cassandra driver 3.x has forward compatibility for server 4.x using V5 protocol as far as I know.

**Comment by robd003** [comment]
I think I'll take a crack at making the trino-scylla plugin. From what I saw in the docs it looked like the 3.x and 4.x driver were very similar with just different package paths.

https://github.com/scylladb/java-driver/tree/scylla-4.x/upgrade_guide

**Comment by ebyhr** [comment]
>  it looked like the 3.x and 4.x driver were very similar with just different package paths.

This is not true. Please take a look at https://github.com/trinodb/trino/pull/7828

**Comment by findepi** [comment]
It sounds like existing Cassandra connector is used with ScyllaDB anyway, but there is room for improvement
@robd003 can you please add a product test environment with ScyllaDB? 
See https://github.com/trinodb/trino/blob/44c385478c3b6d75f7a166b6623ef586095251c6/testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/Suite6NonGeneric.java#L45

**Comment by robd003** [comment]
@findepi Sure, I'll try to get this working in the next few days.

I'll create a separate branch on my fork and then once it's working we can merge the test case in?

**Comment by ebyhr** [comment]
Yes. Please don't forget limiting SMP so that we can run ScyllaDB image in a machine having many cores.

**Comment by findepi** [comment]
cc @haaawk

**Comment by jpprina-dev** [comment]
I need this! 🙏 I would love to see it in future releases. Thanks for considering this enhancement!

**Comment by zseta** [comment]
Well, there's a history of unfinished PRs here unfortunately. I will attempt to work on this, would be nice to have this integration. Looks like there's no _real_ blocker?
